KsApi/models/GraphUser.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+public struct GraphUser: Swift.Decodable {
```

Comment:
Now that the PR with the UserEnvelope is merged I can use it. Regarding the usage of multiple micro structs, I just wonder what would be the complexities involved in case we needed more than one field (like `email` and `currency`, for example. Is this case, how would we decode multiple object types?)

Reply:
I think this is kind of a bigger discussion around how we want to think of our "models" moving forward as we start to use GraphQL more and more. My gut says that for now lets just take it case-by-case and see what makes the most sense: if a new feature were to come up that required both email, currency, and some other user values, then it might make more sense to use the converted `Swift.Decodable` User object to fetch all those fields.

But here's another example: in change email, eventually we have to add the "isEmailVerified" property. I imagine that in that case, since it's still only two properties, we would add that to the struct where `email` lives. But then what do we call this struct? `UserEmail` no longer makes sense, but what could make sense is `UserChangeEmailModel`, ie. the model containing the properties the feature needs in order to function. This starts to move us in a direction where our models are no longer necessarily closely associated with the underlying data model of the overall Kickstarter application, but rather the model required to populate our client's views. From my understanding this is part of the GraphQL design, and something they see as a strength. From the iOS side, we need to decide whether we want to move in that direction, or maintain more "pure" models objects (models like `User`, `Project`, `Reward`, etc) with all optional properties (similar to what @justinswart described as their approach on Drip)

Reply:
Yeah 🤔it's kind've tricky with Swift types. I'm guessing this stuff is easier in JS/TypeScript, but I don't really know.

Another option I was considering could be something like a `UserProtocol` which the various GraphQL types could conform to through extensions.

Also, we could consider really tiny protocols like `UserEmailProtocol` and `UserCurrencyProtocol` and then in a situation where both are required perhaps they can be composed `UserEmailProtocol & UserCurrencyProtocol`.

Although I'm not sure if this would help at all, it might just make _more_ complicated.